go/master/client.go

```diff
@@ -104,25 +103,44 @@ func NewClient(opts ...func(*Client) error) (*Client, error) {
 		if err != nil {
 			return nil, err
 		}
-
 	}
-
+	c.passID = 0
+	c.ch = make(chan record, c.bufSize)
+	// FIXME: connection is created asyncrosly in monitorMaster go routine,
+	//        ensure the connection is ready for use before calling c.addClient.
+	time.Sleep(time.Second)
 	return c, nil
 }
 
+// StartGetRecords must be called at beginning of each pass
+func (c *Client) StartGetRecords(passID int) {
+	c.passID = passID
```

Comment:
This will not actually cause data race because `c.StartGetRecords` will be called only if `getRecords` exits, anyway simply pass `passID` is simpler!